internal/expr/parts.go

```diff
@@ -43,6 +43,10 @@ func (p *inputPart) String() string {
 
 func (p *inputPart) part() {}
 
+func (p *inputPart) isInsert() bool {
```

Comment:
I'm not really sure this extra function in necessary for this PR, we can change it in a general improvements PR in the future, but right now I think we can get away with maybe just adding comments where it is unclear.

Reply:
I had to add a check to see if a given inputPart was part of an insert statement. Although I agree with you it is not strictly related to this PR, I think the effort of copying the check and adding a comment is the same as this small refactor which will make it clearer for all usecases.

Reply:
Fair enough.